src/Microsoft.SqlTools.ServiceLayer/Program.cs

```diff
@@ -49,13 +52,21 @@ internal static async Task Main(string[] args)
                     sqlClientListener = new SqlClientListener();
                 }
 
+
                 // set up the host details and profile paths 
                 var hostDetails = new HostDetails(version: new Version(1, 0));
 
                 SqlToolsContext sqlToolsContext = new SqlToolsContext(hostDetails);
                 ServiceHost serviceHost = HostLoader.CreateAndStartServiceHost(sqlToolsContext);
                 serviceHost.MessageDispatcher.ParallelMessageProcessing = commandOptions.ParallelMessageProcessing;
 
+                if (commandOptions.EnableSqlAuthenticationProvider)
```

Comment:
this doesn't seem to be the typical pattern we're using to get state into a service.  Maybe passing into the service constructor is more scalable then special-casing a particular option in the main method?